### PR TITLE
For modern Chef

### PR DESCRIPTION
**Pull request by limitusus**
This cookbook is depended on by [daemontools cookbook](https://github.com/jtimberman/daemontools), and both of them are using [very old custom resource DSL](https://docs-archive.chef.io/release/12-0/custom_resources.html).
This PR rewrites all the custom resources with the latest one and added tests for them, without changing resource interfaces, except removing `exists` from `group` resource.

Test passes as here: https://app.circleci.com/pipelines/github/limitusus/pacman?branch=feature%2Fchef16

### CODE REVIEW

**Comment by limitusus** [comment]
@jesseadams Could you check this please? Or I release it as a new fork cookbook.

**Comment by jesseadams** [comment]
@limitusus - Would you like for me to pass over ownership of this repo to you? I no longer use chef.

File: resources/aur.rb [hunk 1]
```diff
@@ -17,14 +17,115 @@
...
-attribute :builddir, :default => "#{Chef::Config[:file_cache_path]}/builds"
-attribute :options, :kind_of => String
-attribute :pkgbuild_src, :default => false
-attribute :patches, :kind_of => Array, :default => []
-attribute :exists, :default => false
+property :package_name, String, name_property: true
+property :version, [String, NilClass], default: lazy {
+  v = ''
+  r = ''
+  a = ''
```
Comment (jesseadams) [inline resources/aur.rb]: These variable names should be more descriptive, so you know what they are.

File: test/integration/default/default_test.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+describe package('memb') do
+  it { should be_installed }
+end
+
+describe package('tinydns') do
+  it { should be_installed }
```
Comment (jesseadams) [inline test/integration/default/default_test.rb:6]: should this also confirm version?

File: resources/aur.rb (hunk 1 above)
Comment (jesseadams) [reply resources/aur.rb]: I know you just carried these over though, which I inherited from a different maintainer

Comment (limitusus) [reply resources/aur.rb]: changed to more descriptive variable names

File: test/integration/default/default_test.rb (hunk 2 above)
Comment (limitusus) [reply test/integration/default/default_test.rb:6]: Exactly. I added version check

**Comment by limitusus** [comment]
@jesseadams 
Sure. I will carry it over if you transfer GH repo and Supermarket cookbook to me :)

**Comment by limitusus** [comment]
@jesseadams 
I noticed transferring Supermarket cookbook to me is enough.
I'll update relevant links after that.

**Comment by jesseadams** [comment]
Transfer email sent

**Comment by limitusus** [comment]
Thank you. I received its ownership.
I close this PR and handle this in my repo.

**Pull request closed without merging**